bazelisk.py

```diff
@@ -339,21 +338,30 @@ def download_bazel_into_directory(version, is_commit, directory):
     return destination_path
 
 
-def download(url, destination_path):
-    sys.stderr.write("Downloading {}...\n".format(url))
-    request = Request(url)
-    if "BAZELISK_BASE_URL" in os.environ:
-        parts = urlparse(url)
-        creds = None
+def download(url, destination_path, retries=5, wait_seconds=5):
+    while retries > 0:
+        retries -= 1
         try:
-            creds = netrc.netrc().hosts.get(parts.netloc)
+            sys.stderr.write("Downloading {}...\n".format(url))
+            request = Request(url)
+            if "BAZELISK_BASE_URL" in os.environ:
+                parts = urlparse(url)
+                creds = None
+                try:
+                    creds = netrc.netrc().hosts.get(parts.netloc)
+                except Exception:
+                    pass
+                if creds is not None:
+                    auth = base64.b64encode(("%s:%s" % (creds[0], creds[2])).encode("ascii"))
+                    request.add_header("Authorization", "Basic %s" % auth.decode("utf-8"))
```

Comment:
Can we construct `auth` and `request` once and use them across retries ?

Reply:
I don't see why not. I've made the changes but I'm hitting issues testing it locally. Don't think it's related to the changes, seems it's just unable to download a specific Bazel checksum file; specifically `https://storage.googleapis.com/bazel-builds/artifacts/ubuntu1404/f3aa4184c0b53864f597ecfb8969938e9a2a8287/bazel.sha256`

I'll go ahead and push my changes and then try reverting them locally to see if the old code in this PR works or if this is indeed an issue accessing that resource.

Reply:
As a side note I also added configuration of the retry logic via environment variables to this PR. If you'd me to change those variables or how those settings are sourced just let me know.

Reply:
> I don't see why not. I've made the changes but I'm hitting issues testing it locally. Don't think it's related to the changes, seems it's just unable to download a specific Bazel checksum file; specifically `https://storage.googleapis.com/bazel-builds/artifacts/ubuntu1404/f3aa4184c0b53864f597ecfb8969938e9a2a8287/bazel.sha256`
> 
> I'll go ahead and push my changes and then try reverting them locally to see if the old code in this PR works or if this is indeed an issue accessing that resource.

Figured it out, didn't realize that for the checksum the outer scope was expecting a 404 `HTTPError` to be thrown if the checksum doesn't exist for the specific Bazel version.